### PR TITLE
Add Attribute Picker as a selector - add to state trigger

### PR DESCRIPTION
**Pull request by zsarnett**
## Proposed change

image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/ha-form/ha-form.ts [hunk 1]
```diff
@@ -136,7 +137,7 @@ export class HaForm extends LitElement implements HaFormElement {
         display: block;
       }
       .root > *:not([own-margin]) {
-        margin-bottom: 24px;
+        margin-bottom: 16px;
```
Comment (balloob) [inline src/components/ha-form/ha-form.ts:140]: I am pretty sure that this was aligned with the "own-margin" components margin below. We should test this better.

Comment (balloob) [reply src/components/ha-form/ha-form.ts:140]: I think it was driven by the helper text of mwc-textfield that hides when not in focus but still takes up space.

File: src/components/ha-selector/ha-selector-attribute.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import "../entity/ha-entity-attribute-picker";
```
Comment (balloob) [inline src/components/ha-selector/ha-selector-attribute.ts:1]: Add to gallery pages for forms and selectors.

File: src/components/ha-form/ha-form.ts [hunk 3]
```diff
@@ -81,6 +81,7 @@ export class HaForm extends LitElement implements HaFormElement {
               : ""}
             ${"selector" in item
               ? html`<ha-selector
+                  .schema=${item}
```
[the remark this answers is not in the document]
Comment (balloob) [reply src/components/ha-form/ha-form.ts:84]: Adding features outside of the `selector` definition will break compatibility when selectors are used in `services.yaml` or blueprints.

[the remark this answers is not in the document]
Comment (zsarnett) [reply src/components/ha-form/ha-form.ts:84]: When the value is changed and sent to the ha-form. Ha-form uses schema to find the name of the value

[the remark this answers is not in the document]
Comment (zsarnett) [reply src/components/ha-form/ha-form.ts:84]:
```ts
root.addEventListener("value-changed", (ev) => {
      ev.stopPropagation();
      const schema = (ev.target as HaFormElement).schema as HaFormSchema;
      fireEvent(this, "value-changed", {
        value: { ...this.data, [schema.name]: ev.detail.value },
      });
    });
```

[the remark this answers is not in the document]
Comment (balloob) [reply src/components/ha-form/ha-form.ts:84]: Then how does the selectors card on https://design.home-assistant.io/#components/ha-form work?

[the remark this answers is not in the document]
Comment (zsarnett) [reply src/components/ha-form/ha-form.ts:84]: Is it the latest dev?

[the remark this answers is not in the document]
Comment (zsarnett) [reply src/components/ha-form/ha-form.ts:84]: I broke it today

**Review by balloob — APPROVED** [review]

File: src/panels/config/automation/trigger/types/ha-automation-trigger-state.ts [hunk 4]
```diff
@@ -81,12 +92,13 @@ export class HaStateTrigger extends LitElement implements TriggerElement {
     const trgFor = createDurationData(this.trigger.for);
 
     const data = { ...this.trigger, ...{ for: trgFor } };
```
Comment (bramkragten) [inline src/panels/config/automation/trigger/types/ha-automation-trigger-state.ts:83]: { ...this.trigger, for: trgFor }

**Pull request merged**